src/app/modules/mod.rs

```diff
@@ -31,11 +33,7 @@ define_error! {
     }
 }
 
-/// Module trait
-pub(crate) trait Module {
-    /// The module's Store type
-    type Store;
-
+pub(crate) trait Module<S: Store>: Send + Sync {
```

Comment:
what motivated the change of `Store` being an associated type to now being a trait bound for the generic `S`?

Reply:
I think was an oversight on my part during the merge with the typed-store branch. Fixed here -> https://github.com/informalsystems/basecoin-rs/pull/44/commits/ff29c12a7ceb00a0637483565fa6d2b9a4909a6b